paf-mvp-frontend/tests/test-cases/queue.test.ts

```diff
@@ -0,0 +1,58 @@
+import { TestWatcher } from 'jest';
+import { isExportDeclaration } from 'typescript';
+import { Command, CommandQueue, processCommands } from '../../src/utils/queue';
+
+describe('queue', () => {
+  let cmd1: jest.Mock;
+  let cmd2: jest.Mock;
+  let cmd3: jest.Mock;
+  const lib = {
+    queue: [] as CommandQueue,
+  };
+
+  beforeEach(() => {
+    cmd1 = jest.fn();
+    cmd2 = jest.fn();
+    cmd3 = jest.fn();
+    lib.queue = [];
+  });
+
+  test('processCommands undefined do not crash', () => {
+    lib.queue = processCommands(undefined);
+  });
+
+  test('processCommands empty array do nothing', () => {
+    lib.queue = processCommands(lib.queue);
+  });
+
+  test('processCommands 2 commands', () => {
+    lib.queue = processCommands([cmd1, cmd2]);
+
+    expect(cmd1.mock.calls.length).toBe(1);
+    expect(cmd2.mock.calls.length).toBe(1);
+  });
+
+  test('returned processor process correctly', () => {
+    lib.queue = processCommands(lib.queue);
+    lib.queue.push(cmd1, cmd2);
+    lib.queue.push(cmd3);
+
+    expect(cmd1.mock.calls.length).toBe(1);
+    expect(cmd2.mock.calls.length).toBe(1);
+    expect(cmd3.mock.calls.length).toBe(1);
+  });
+
+  test('documented setup', () => {
+    // directly in page
+    lib.queue.push(cmd1);
+    lib.queue.push(cmd2);
+
+    // in asynchronous script
+    lib.queue = processCommands(lib.queue);
```

Comment:
This is an illustration of what I dislike about this approach.
This method does two things: it actually processes commands as expected, but also returns a new object that callers must re-affect to `lib.queue`, which is IMHO, confusing.

Reply:
First, I would like to point out that I don't think that this amount of comments is productive for the project. To be discussed informally (again) but IMO, you should adapt your cursor.

I think it makes sense: a one-liner that you copy-paste from the doc is a no-brainer and removes possibilities of errors.